### ISSUE TITLE
`Get-Help`'s completion cannot complete `about_` help topics of 3rd-party modules, even thought they are recognised as such

### ISSUE THREAD

**Issue by matteocoder** [PowerShell/PowerShell#20648]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Prerequisite: use a `profile.ps1` like this:
```ps1
if (([console]::outputEncoding).EncodingName -ne 'Unicode (UTF-8)') {
    [Console]::InputEncoding = [Console]::OutputEncoding = $OutputEncoding = [System.Text.Utf8Encoding]::new()
}
import-module posh-git
set-culture en-US
set-psreadlineoption -predictionsource none
```

1. `Install-Module posh-git -scope currentuser -force`
2. `update-help`
3. `Import-Module posh-git`
4. `Get-Help about_*` (to check if `about_posh-git` is included among the help topics)
5. `Get-Help about_po<ctrl+space>`

Note: `posh-git` is used as an example in this case, but this bug manifests even with other modules such as `PSFzf`. Both modules have been installed from the Powershell Gallery.

Additionally, my $USERPROFILE directory contains a space in the path; perhaps that is relevant.

### Expected behavior

```console
PS> get-help about_po<ctrl+space>
about_PowerShell_Editions about_PowerShell_Config about_posh-git
```

### Actual behavior

```console
PS> get-help about_po<ctrl+space>
about_PowerShell_Editions about_PowerShell_Config
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.0-rc.1
PSEdition                      Core
GitCommitId                    7.4.0-rc.1
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![helptabcomp](https://github.com/PowerShell/PowerShell/assets/22897138/733045ef-15fe-4227-bab1-7530cea8273c)

**Comment by MartinGC94** [comment]
This is because it never searches for help in the module directories, see: https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs#L7574 the module paths just needs to be added to the list of directories to scan.

**Comment by matteocoder** [comment]
@MartinGC94 Thank you for your reply. How do you add the module directories to the list of directories to scan?

**Comment by matteocoder** [comment]
@MartinGC94 Ah, I see. Thanks again.

**Comment by MartinGC94** [comment]
I think this is working now and that it was fixed by https://github.com/PowerShell/PowerShell/pull/24194  
Can you check? I at least see help topics from PSReadline and other modules.

**Comment by matteocoder** [comment]
@MartinGC94 Yes, it works now. Thank you all for your work to make PowerShell a great shell environment!
